src/adapters/mc/OsConfigResource.c

```diff
@@ -539,7 +567,11 @@ void MI_CALL OsConfigResource_Invoke_GetTargetResource(
     if (MI_TRUE == isCompliant)
     {
         reasonCode = DuplicateString(passCode);
-        reasonPhrase = DuplicateString(auditPassed);
+        if ((0 == strcmp(g_reportedObjectValue, g_passValue)) || 
+            ((strlen(g_reportedObjectValue) > strlen(g_passValue)) && (NULL == (reasonPhrase = DuplicateString(g_reportedObjectValue + strlen(g_passValue))))))
+        {
+            reasonPhrase = DuplicateString(auditPassed);

```

Comment:
Can we use `strdup()` here instead?

https://man7.org/linux/man-pages/man3/strdup.3.html

Reply:
We have our own and I do prefer it as a match to FormatAllocateString.

Reply:
What's the difference between our own and strdup?